packages/calcite-components/src/components/swatch/swatch.scss

```diff
@@ -0,0 +1,122 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-swatch-corner-radius: Specifies the component's corner radius.
+ *
+ */
+
+// AUTO-GENERATED — do not modify. Changes will be overwritten.
+//
+// Internal CSS custom properties for component use only. Overwriting is not recommended.
+//
+// --calcite-internal-swatch-inset
+// --calcite-internal-swatch-size
+
+:host {
+  @apply block;
+  --calcite-internal-swatch-inset: var(--calcite-spacing-xxs);
+}
+
+:host([scale="s"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xl);
+  }
+}
+
+:host([scale="m"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxl);
+  }
+}
+
+:host([scale="l"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxxl);
+  }
+}
+
+.container {
+  @apply relative focus-base justify-center box-border  overflow-hidden;
+  font-size: var(--calcite-internal-swatch-font-size, var(--calcite-font-size));
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+
+  &:not(.non-interactive):hover {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):hover {
+    box-shadow: 0 0 0 2px var(--calcite-color-border-2);
+  }
+  &.selectable {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):focus {
+    @apply focus-normal;
+  }
+}
+
+.swatch {
+  position: absolute;
+  @apply inline-flex overflow-hidden absolute z-10;
```

Comment:
should the z-index use one of our zIndex css vars?

Reply:
Good q - the goal is just to create an internal "local" stack - so using a semantic token feels weird - but then our core set is missing "200" https://developers.arcgis.com/calcite-design-system/foundations/tokens/reference/#stacking-1.

These TW ones offer smaller increments which is nice.

Reply:
I've used this pattern in the shell component:

```scss
z-index: calc(var(--calcite-z-index) + 1);
```

That way we can use a token and add values on top of it to handle local internal stacking.

Reply:
Nice, updated to use that pattern.